src/main/resources/js/view.context.menu.js

```diff
@@ -67,18 +66,19 @@ function setUpDialogForCreateAction(id) {
 			};
 		}
 	});
+
+	setUpDialog();
 }
 
 function setUpDialog() {
-	resetDialog();
-	AJS.dialog2("#dialog").show();
-	AJS.dialog2("#dialog").on("show", function() {
-		resetDialog();
-	});
-	AJS.$(document).on("click", "#dialog-cancel-button", function(e) {
-		e.preventDefault();
-		AJS.dialog2("#dialog").hide();
+	document.getElementById("dialog-cancel-button").addEventListener("click", function() {
+		closeDialog();
 	});
+	AJS.dialog2("#dialog").show();
+	AJS.dialog2("#dialog").hide();
+	AJS.dialog2("#dialog").show();
+	AJS.dialog2("#dialog").hide();
+	AJS.dialog2("#dialog").show();
```

Comment:
@lw2011 and @Jcl15 Unfortunately, I could not manage to get rid of this call sequence using callback functions or promise. I tried the following but this does not work:
`$.when(setUpDialogForCreateAction(id)).done(function() {
            AJS.dialog2("#dialog").show();
        });`
Could you please try to solve this in code and commit it?